File: digital-root/digitalRoot.js
const digitalRoot = (number) => {

    const sum = (num) => {
        let output = 0;
        num.toString().split('').forEach((element) => {
            output += Number(element);
        });
        return output;
    };

    let initialSum = sum(number);
    if(initialSum < 9){
        return initialSum;
    };

    let secondSums = 0;
    do {
        secondSums = sum(initialSum);
        if(secondSums > 9){
            secondSums = sum(secondSums);
        };
    } while (secondSums > 9);

    return secondSums;
};

module.exports = { digitalRoot };

//https://www.codewars.com/kata/541c8630095125aba6000c00/train/javascript

File: python-venv/digital_root/kata_function/digital_root.py
def digital_root(integer):

    def equate(num):
        toString = list(str(num))
        toInt = [int(numString) for numString in toString]
        return sum(toInt)
    
    initialSum = equate(integer)

    if initialSum < 9:
        return initialSum
    
    secondEquate = 10
    while secondEquate > 9:
        secondEquate = equate(initialSum)
        if secondEquate > 9:
            secondEquate = equate(secondEquate)

    return secondEquate

#https://www.codewars.com/kata/541c8630095125aba6000c00/train/python